src/test/java/net/emustudio/edigen/templates/DisassemblerTest.java

```diff
@@ -34,4 +49,42 @@ public void testAbsoluteStrategyWorks() {
         }
         return result;
     }
+
+    public static byte[] bit_reverse(byte[] value) {
+        byte[] result = new byte[value.length];
+
+        for (int octet = 0; octet < result.length; octet++) {
+            for (int bit = 0; bit < 8; bit++) {
+                result[octet] |= (value[octet] & (1 << bit)) >>> bit << (8 - bit - 1);
+            }
+        }
+
+        return result;
+    }
+
+    public static byte[] shift_left(byte[] value) {
+        byte[] result = new byte[value.length];
+
+        for (int octet = 0; octet < result.length; octet++) {
+            int shifted = (value[octet] << 1) & 0xFF;
+            if (octet + 1 < result.length) {
+                shifted |= value[octet + 1] >>> 7;
+            }
+            result[octet] = (byte)shifted;
+        }
+        return result;
+    }
+
+    public static byte[] shift_right(byte[] value) {
+        byte[] result = new byte[value.length];
+
+        for (int octet = result.length - 1; octet >= 0; octet--) {
+            int shifted = (value[octet] >>> 1) & 0xFF;
+            if (octet - 1 >= 0) {
+                shifted |= ((value[octet - 1] & 1) << 7);
+            }
+            result[octet] = (byte)shifted;
+        }
+        return result;
+    }
```

Comment:
Copying methods from the disassembler template is not ideal. As a quick fix, I can imagine creating a separate Java file with decoding strategies in Edigen, which would be copied verbatim to the resulting disassembler.

Reply:
I agree with you, this is not ideal. However, copying whole class to disassembler I also see as not ideal, because it's quite complicated. But what I see as quick-fix and kind-of win-win is place the strategy in emuLib. Currently we already have there `DecodedInstruction` class and the class `NumberUtils` already contains some operations which allows use big endian/little endian/reverse strategy... Maybe if this class will be refactored, then edigen's decoding Strategy can be used in both places.

Reply:
In addition I found that the `DecodedInstruction` uses internally it's own `reverseBytes` method... Which is exactly the same as `little_endian()` strategy